File: app/src/main/kotlin/net/transcode001/creambox/TwitterUtils.java
package net.transcode001.creambox;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.AsyncTask;

import java.util.concurrent.ExecutionException;

import twitter4j.Twitter;
import twitter4j.TwitterFactory;
import twitter4j.auth.AccessToken;
import twitter4j.conf.Configuration;
import twitter4j.conf.ConfigurationBuilder;

public class TwitterUtils {

    public static Twitter getInstance(Context context){
        String consumerKey = context.getString(R.string.consumerKey);
        String consumerSecret = context.getString(R.string.consumerSecret);

        TwitterFactory tf = new TwitterFactory();
        Twitter twitter = tf.getInstance();
        twitter.setOAuthConsumer(consumerKey,consumerSecret);

        if (hasAccessToken(context)) {
            twitter.setOAuthAccessToken(loadAccessToken(context));
        }

        return twitter;

    }

    public static Configuration getConfigurationInstance(Context context){
        String consumerKey = context.getString(R.string.consumerKey);
        String consumerSecret = context.getString(R.string.consumerSecret);
        String accessToken = context.getString(R.string.accessToken);
        String accessTokenSec = context.getString(R.string.accessTokenSec);

        return new ConfigurationBuilder().setOAuthConsumerKey(consumerKey)
                .setOAuthConsumerSecret(consumerSecret).setOAuthAccessToken(accessToken)
                .setOAuthAccessTokenSecret(accessTokenSec).build();
    }

    public static Twitter getAccessTokenInstance(Twitter mTwitter,Context context){
        String accessToken = context.getString(R.string.accessToken);
        String tokenSecret = context.getString(R.string.accessTokenSec);
        AccessToken at = new AccessToken(accessToken,tokenSecret);
        mTwitter.setOAuthAccessToken(at);
        return mTwitter;
    }

    /**
     * AccessTokenを永続的に保存
     * getSharedPreference("設定データの名前",ファイルの操作モード)
     *
     */
    public static void storeAccessToken(Context context, AccessToken accessToken){
        SharedPreferences preferences = context.getSharedPreferences("pref_name",
                Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString("token", accessToken.getToken());
        editor.putString("token_secret", accessToken.getTokenSecret());
        editor.apply();
    }

        public static AccessToken loadAccessToken(Context context) {
            final Context mContext = context;
            SharedPreferences preferences = mContext.getSharedPreferences("pref_name", Context.MODE_PRIVATE);
            String token = preferences.getString("token", null);
            String tokenSecret = preferences.getString("token_secret", null);
            if (token != null && tokenSecret != null) {
                return new AccessToken(token, tokenSecret);
            } else {
                return null;
            }
        }

    public static boolean hasAccessToken(Context context) {
        return loadAccessToken(context) != null;
    }

}

File: app/src/main/java/net/transcode001/creambox/TwitterUtils.kt
package net.transcode001.creambox

import android.content.Context
import android.content.SharedPreferences
import android.os.AsyncTask

import java.util.concurrent.ExecutionException

import twitter4j.Twitter
import twitter4j.TwitterFactory
import twitter4j.auth.AccessToken
import twitter4j.conf.Configuration
import twitter4j.conf.ConfigurationBuilder

object TwitterUtils {

    fun getInstance(context: Context): Twitter {
        val consumerKey = context.getString(R.string.consumerKey)
        val consumerSecret = context.getString(R.string.consumerSecret)

        val tf = TwitterFactory()
        val twitter = tf.instance
        twitter.setOAuthConsumer(consumerKey, consumerSecret)

        if (hasAccessToken(context)) {
            twitter.oAuthAccessToken = loadAccessToken(context)
        }

        return twitter

    }

    fun getConfigurationInstance(context: Context): Configuration {
        val consumerKey = context.getString(R.string.consumerKey)
        val consumerSecret = context.getString(R.string.consumerSecret)
        val accessToken = context.getString(R.string.accessToken)
        val accessTokenSec = context.getString(R.string.accessTokenSec)

        return ConfigurationBuilder().setOAuthConsumerKey(consumerKey)
                .setOAuthConsumerSecret(consumerSecret).setOAuthAccessToken(accessToken)
                .setOAuthAccessTokenSecret(accessTokenSec).build()
    }

    fun getAccessTokenInstance(mTwitter: Twitter, context: Context): Twitter {
        val accessToken = context.getString(R.string.accessToken)
        val tokenSecret = context.getString(R.string.accessTokenSec)
        val at = AccessToken(accessToken, tokenSecret)
        mTwitter.oAuthAccessToken = at
        return mTwitter
    }

    /**
     * AccessTokenを永続的に保存
     * getSharedPreference("設定データの名前",ファイルの操作モード)
     *
     */
    fun storeAccessToken(context: Context, accessToken: AccessToken) {
        val preferences = context.getSharedPreferences("pref_name",
                Context.MODE_PRIVATE)
        val editor = preferences.edit()
        editor.putString("token", accessToken.token)
        editor.putString("token_secret", accessToken.tokenSecret)
        editor.apply()
    }

    fun loadAccessToken(context: Context): AccessToken? {
        val preferences = context.getSharedPreferences("pref_name", Context.MODE_PRIVATE)
        val token = preferences.getString("token", null)
        val tokenSecret = preferences.getString("token_secret", null)
        return if (token != null && tokenSecret != null) {
            AccessToken(token, tokenSecret)
        } else {
            null
        }
    }

    fun hasAccessToken(context: Context): Boolean {
        return loadAccessToken(context) != null
    }

}
